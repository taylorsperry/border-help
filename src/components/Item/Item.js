import React, { Component } from 'react'

export class Item extends Component {
  constructor(props) {
    super(props);
    this.state = {
      display: false
    }
  }

  toggleDetail = () => {
    this.setState({
      display: !this.state.display
    })
  }

  render() {
    const { title, detail } = this.props.item
    
    let displayDetail
    
    if (this.state.display === true) {
      displayDetail = detail
    } 

    return (
      <div>
        <h2 onClick={this.toggleDetail}>{title}</h2>
        {displayDetail}
      </div>
    )
  }
}

export default Item