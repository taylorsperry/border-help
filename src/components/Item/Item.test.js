import React from 'react'
import { Item } from './Item'
import { shallow } from 'enzyme'

describe('Item', () => {
  let wrapper
  let mockProps = {
    key: 1,
    item: {
      title: 'title',
      detail: ['detail', 'two']
    }
  }
  beforeEach(() => {
    wrapper=shallow(<Item {...mockProps}/>)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    let expected = {
      display: false
    }
    expect(wrapper.state()).toEqual(expected)
  })

  it('should toggle the display', () => {
    let expected = {
      display: false
    }

    let mockEvent = {target: {}}

    let newState = {
      display: true
    }

    wrapper.instance()
    expect(wrapper.state()).toEqual(expected)

    wrapper.instance().toggleDetail(mockEvent)
    expect(wrapper.state()).toEqual(newState)
  })
})