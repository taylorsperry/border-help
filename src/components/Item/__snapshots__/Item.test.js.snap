// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match the snapshot 1`] = `
<div
  className="item"
>
  <h3
    className="title"
    onClick={[Function]}
  >
    title
  </h3>
</div>
`;
